src/utilities/string_functions.js

```diff
@@ -295,20 +295,23 @@ p5.prototype.nf = function(nums, left, right) {
 };
 
 function doNf(num, left, right) {
+  let isNegative = num < 0;
+  num = Math.abs(num);
   let [leftPart, rightPart] = num.toString().split('.');
 
+
   if (typeof right === 'undefined') {
     leftPart = leftPart.padStart(left, '0');
-    return rightPart ? leftPart + '.' + rightPart : leftPart;
+    let result = rightPart ? leftPart + '.' + rightPart : leftPart;
+    return isNegative ? '-' + result : result;
   } else {
-    let roundedOff = num.toFixed(right);
-    [leftPart, rightPart] = roundedOff.toString().split('.');
+    let roundedNum = Math.round(num * Math.pow(10, right));
+    roundedNum = roundedNum / Math.pow(10, right);
+    [leftPart, rightPart] = roundedNum.toString().split('.');
     leftPart = leftPart.padStart(left, '0');
-    if(typeof rightPart === 'undefined'){
-      return leftPart;
-    }else{
-      return leftPart + '.' + rightPart;
-    }
+    let result = typeof rightPart === 'undefined' ?
```

Comment:
Do you think @Akhilbisht798 that using ternary operator for result could be a bit complicated for the other contributors to understand in a first look. Can we use if and else block for this part only? Do you agree, or maybe you have any other thoughts? Please do let me know.

Reply:
I have no problem in using if else block for this statement if that increase the look. Let me known what you think about the other changes and I will commit the changes of this with them.

Reply:
just a thought...  

if we want to reduce the usage of ternary operator it can be mentioned in the contributor guideline somewhere

or if we have strong opinions about it, it should be part of some lint system to give warning / error

Reply:
> just a thought...
> 
> if we want to reduce the usage of ternary operator it can be mentioned in the contributor guideline somewhere
> 
> or if we have strong opinions about it, it should be part of some lint system to give warning / error

It's not about reducing the use of ternary operators. In this case, I was just considering whether using if and else blocks might make the code more readable, especially when checking the type. If it's undefined, we perform one operation, and for everything else, we do something different—all in a single line. I'm open to discussion and would love to hear your thoughts on this. Otherwise ternary operator also works for me. Any feedback @Akhilbisht798 and @ashish1729 ? 

```
let result = typeof rightPart === 'undefined' ? leftPart : leftPart + '.' + rightPart;
```